File: Chef and Strings.py
try:
    t=int(input())
    while t:
        add=0
        inp=[]
        size=int(input())
        inp=input().split()
        temp=int(inp[0])
        for i in inp[1:size]:
            if temp!=int(i):
                add+=abs(temp-int(i))-1
            temp=int(i)
        print(add)
except:
    pass

File: Chef and Strings.cpp
#include <iostream>
using namespace std;

int main() {
	int t;
    cin >> t;
    while(t--)
    {
        long temp=0,temp1=0,size,sum=0;
        cin >> size;
        cin >> temp;
        while(--size)
        {
            cin >> temp1;
            sum+=abs(temp-temp1)-1;
            temp=temp1;
        }
        cout << sum << endl;
    }
	return 0;
}
